lib/router/handler-info.js

```diff
@@ -238,40 +227,54 @@ HandlerInfo.prototype = {
       (this.hasOwnProperty('context') && !contextsMatch) ||
       (this.hasOwnProperty('params') && !paramsMatch(this.params, other.params))
     );
-  },
-};
+  }
 
-Object.defineProperty(HandlerInfo.prototype, 'handler', {
-  get: function() {
+  get handler() {
     // _handler could be set to either a handler object or undefined, so we
     // compare against a default reference to know when it's been set
     if (this._handler !== DEFAULT_HANDLER) {
       return this._handler;
     }
 
     return this.fetchHandler();
-  },
+  }
 
-  set: function(handler) {
+  set handler(handler) {
     return (this._handler = handler);
-  },
-});
+  }
 
-Object.defineProperty(HandlerInfo.prototype, 'handlerPromise', {
-  get: function() {
-    if (this._handlerPromise) {
+  get handlerPromise() {
+    if (this._handlerPromise !== null) {
       return this._handlerPromise;
     }
 
     this.fetchHandler();
 
     return this._handlerPromise;
-  },
+  }
 
-  set: function(handlerPromise) {
-    return (this._handlerPromise = handlerPromise);
-  },
-});
+  set handlerPromise(handlerPromise) {
+    this._handlerPromise = handlerPromise;
+
+    return handlerPromise;
+  }
+}
+
+// this is bonkers, we require that `context` be set on on the
+// HandlerInfo prototype to null because the checks in
+// `NamedTransitionIntent.prototype.applyToHandlers` here
+// https://github.com/tildeio/router.js/blob/v1.2.8/lib/router/transition-intent/named-transition-intent.js#L76-L81
+// check of `oldHandlerInfo.context === newHandlerInfo.context` and assumes
+// that the params _must_ match also in that case.
+//
+// The only reason `oldHandlerInfo.context` and `newHandlerInfo.context` did not
+// match in prior versions is because if the context isn't set yet (on newHandlerInfo)
+// is because it inherits the `null` from the prototype vs `undefined` (on
+// the oldHandlerInfo).
+//
+// A future refactoring should remove that conditional, and fix the hand full of
+// failing tests.
+HandlerInfo.prototype.context = null;
```

Comment:
Made an issue to track this future cleanup/fixing here: https://github.com/tildeio/router.js/issues/225.